addons/web/controllers/pivot.py

```diff
@@ -19,55 +19,71 @@ def check_xlwt(self):
     @http.route('/web/pivot/export_xls', type='http', auth="user")
     def export_xls(self, data, token):
         jdata = json.loads(data)
-        nbr_measures = jdata['nbr_measures']
         workbook = xlwt.Workbook()
         worksheet = workbook.add_sheet(jdata['title'])
         header_bold = xlwt.easyxf("font: bold on; pattern: pattern solid, fore_colour gray25;")
         header_plain = xlwt.easyxf("pattern: pattern solid, fore_colour gray25;")
         bold = xlwt.easyxf("font: bold on;")
 
-        # Step 1: writing headers
-        headers = jdata['headers']
+        measure_count = jdata['measure_count']
+        origin_count = jdata['origin_count']
+
+        # Step 1: writing col group headers
+        col_group_headers = jdata['col_group_headers']
 
         # x,y: current coordinates
         # carry: queue containing cell information when a cell has a >= 2 height
         #      and the drawing code needs to add empty cells below
         x, y, carry = 1, 0, deque()
-        for i, header_row in enumerate(headers):
+        for i, header_row in enumerate(col_group_headers):
             worksheet.write(i, 0, '', header_plain)
             for header in header_row:
                 while (carry and carry[0]['x'] == x):
                     cell = carry.popleft()
-                    for i in range(nbr_measures):
+                    for i in range(measure_count * (2 * origin_count - 1)):
```

Comment:
Redefining `i` here is bug-prone: use `j`.

Reply:
ok